spec/datadog/core/configuration/base_spec.rb

```diff
@@ -142,6 +142,81 @@
         end
       end
 
+      describe '#default_option?' do
+        let(:configuration) do
+          Class.new do
+            include Datadog::Core::Configuration::Base
+
+            settings :fake_test do
+              option :enabled, default: true
+              option :without_default
+            end
+          end.new
+        end
+
+        context 'with default option' do
+          context 'when not set' do
+            it 'returns true' do
+              expect(configuration.fake_test.default_option?(:enabled)).to be(true)
+            end
+          end
+
+          context 'when not set but accessed' do
+            it 'returns true' do
+              configuration.fake_test.enabled
+              expect(configuration.fake_test.default_option?(:enabled)).to be(true)
+            end
+          end
+
+          context 'when set' do
+            it 'returns false' do
+              configuration.fake_test.enabled = false
+              expect(configuration.fake_test.default_option?(:enabled)).to be(false)
+            end
+          end
+
+          context 'when set and reset' do
+            it 'returns false' do
+              configuration.fake_test.enabled = false
+              expect(configuration.fake_test.default_option?(:enabled)).to be(false)
+              configuration.fake_test.reset!
+              expect(configuration.fake_test.default_option?(:enabled)).to be(true)
+            end
+          end
+        end
+
+        context 'without default option' do
+          context 'when not set' do
+            it 'returns true' do
+              expect(configuration.fake_test.default_option?(:without_default)).to be(false)
```

Comment:
The test description doesn't match the assertion :)

Reply:
Ahh the wonder of copy pasta 🤣